src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -32,27 +21,27 @@ public LRUCache(int cap) {
 
     private void setCapacity(int newCapacity) {
         checkCapacity(newCapacity);
-        for (int i = data.size(); i > newCapacity; i--) {
-            Entry<K, V> evicted = evict();
-            data.remove(evicted.getKey());
-        }
         this.cap = newCapacity;
+        
     }
 
     private Entry<K, V> evict() {
         if (head == null) {
-            throw new RuntimeException("cache cannot be empty!");
+            throw new RuntimeException("Cache cannot be empty!");
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rishabhbohra3/Algo-tutorial-test/pull/6#issuecomment-2196728573>#cbf3aa</a><span> - 06/28/2024, 11:53 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'evict' method throws a 'RuntimeException' when the cache is empty. This can cause the entire application to crash if not handled properly. This is a critical issue as it affects the stability of the application.
</div>
<div id="fix">
<b>Fix:</b> Instead of throwing a 'RuntimeException', handle the empty cache scenario gracefully. For example, return 'null' or an optional value to indicate that no entry was evicted.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -29,7 +29,7 @@
      if (head == null) {
 -        throw new RuntimeException("Cache cannot be empty!");
 +        return null; // or Optional.empty() if using Java 8+
      }
      Entry<K, V> evicted = head;
      head = evicted.getNextEntry();
```
</div>
</details>
</div>
<br>


</div>